calculate.js

```diff
@@ -9,88 +9,122 @@
+                            order.disbursementDetails[0].brokerageAssistFee+
+                            order.disbursementDetails[0].secFee;
+
+            console.info("Order: ",order.soldQty,"*",order.executedPrice)
+
+            // for all grants in this transaction
+            for (var grant of order.purchaseGrantDetails) {
+
+                var sharesCount = grant.sharesSellSold
+
+                orderGrossRevenue += sellingPricePerShare * sharesCount;
+
+                var RSU = (grant.type == "RSU")
+                var date = grant.vestDate;
+                if (!RSU){
+                    date = grant.purchaseDate;
+                }
+                var dateObj = new Date(date);
+
+                var stockPrice = 0;
+
+                if (RSU || (!RSU && (dateObj <= new Date('12/31/2011') || dateObj >= new Date('12/31/2013')))) {
+                    // if RSU or ESPP, but before 2012 or after 2013
+                    // use the market value
+                    
+
+                    do {
+                        stockPrice = adobeStockValues[date] || 0;
+                        if (stockPrice > 0) break;
+                        console.debug("\t\t go back 1 day and get close price", date, stockPrice)
+                        // go back 1 day nasdaq was closed  
+                        var dateObj = new Date(dateObj.getTime()-24*60*60*1000);
+                        date = dateObj.toLocaleString('en-us', {year: 'numeric', month: '2-digit', day: '2-digit'})
+                    } while(stockPrice == 0)
+                    
+                } else {
+                    // if ESPP between and taxes were not paid on the diff
+                    // use the grant purchase value
+                    stockPrice = grant.purchasePrice;
+                }
+
+                var grantCost = stockPrice * sharesCount;
+                orderCost += grantCost
+                console.info("\t",RSU?"[RSU ]":"[ESPP]","grant cost: ",grantCost,"=",stockPrice,"*",sharesCount)
             }
-
-            var grantCost = stockPrice * sharesCount;
-            orderCost += grantCost
-            console.info("\t",RSU?"[RSU ]":"[ESPP]","grant cost: ",grantCost,"=",stockPrice,"*",sharesCount)
+            var orderProfit = orderGrossRevenue-orderCost-lineComission
+            console.info("order profit: ",orderProfit," = ",orderGrossRevenue,"-",orderCost,"-",lineComission )
+            totalProfits += orderProfit
         }
-        var orderProfit = orderGrossRevenue-orderCost-lineComission
-        console.info("order profit: ",orderProfit," = ",orderGrossRevenue,"-",orderCost,"-",lineComission )
-        totalProfits += orderProfit
-    }
 
-    totalProfits = parseInt(totalProfits*100)/100;
-    console.info("total profit ", totalProfits)
+        totalProfits = parseInt(totalProfits*100)/100;
+        console.info("total profit ", totalProfits);
 
+        continuation();
+    }, 5000);
```

Comment:
Hardcoded value to wait for all the rows to expand (it's not instant, they may do XHR behind the scenes).